app/services/data_exports/export_resources_service.rb

```diff
@@ -1,6 +1,54 @@
 module DataExports
   class ExportResourcesService < BaseService
+    EXPIRED_FAILURE_MESSAGE = 'Data Export already expired'
+    PROCESSED_FAILURE_MESSAGE = 'Data Export already processed'
```

Comment:
Right now I believe we only set error codes and the frontend handles the proper message. We'd need to handle translations otherwise

Reply:
with a quick search in the code I have found other places where the message is set....

```
➜  api git:(feat-export-invoices-staging) ✗ ag result.service_failure!
app/services/invoices/payments/adyen_service.rb
83:        result.service_failure!(code: adyen_error.code, message: adyen_error.msg) unless adyen_success
93:        result.service_failure!(code: e.code, message: e.msg)

app/services/invoices/create_invoice_subscription_service.rb
18:        return result.service_failure!(

app/services/auth/google_service.rb
88:      result.service_failure!(code: 'google_auth_missing_setup', message: 'Google auth is not set up')

app/services/charges/apply_pay_in_advance_charge_model_service.rb
15:        return result.service_failure!(code: 'apply_charge_model_error', message: 'Charge is not pay_in_advance')

app/services/data_exports/export_resources_service.rb
13:      return result.service_failure!(code: 'data_export_expired', message: EXPIRED_FAILURE_MESSAGE) if data_export.expired?
14:      return result.service_failure!(code: 'data_export_processed', message: PROCESSED_FAILURE_MESSAGE) unless data_export.pending?
33:      result.service_failure!(code: e.message, message: e.full_message)

app/services/integrations/aggregator/contacts/update_service.rb
35:          result.service_failure!(code:, message:)

app/services/integrations/aggregator/contacts/create_service.rb
36:          result.service_failure!(code:, message:)

app/services/users_service.rb
81:    result.service_failure!(code: 'token_encoding_error', message: e.message)

app/services/credit_notes/refunds/adyen_service.rb
53:          result.service_failure!(code: 'refund_failed', message: 'Refund failed to perform')

app/services/credit_notes/refunds/gocardless_service.rb
57:          result.service_failure!(code: 'refund_failed', message: 'Refund failed to perform')

app/services/credit_notes/refunds/stripe_service.rb
53:          result.service_failure!(code: 'refund_failed', message: 'Refund failed to perform')

app/services/payment_providers/find_service.rb
28:        return result.service_failure!(
37:        return result.service_failure!(code: 'payment_provider_not_found', message: 'Payment provider not found')

app/services/payment_providers/gocardless_service.rb
42:      result.service_failure!(code: 'internal_error', message: e.description)
65:      result.service_failure!(code: 'webhook_error', message: 'Invalid payload')
67:      result.service_failure!(code: 'webhook_error', message: 'Invalid signature')

app/services/payment_providers/adyen_service.rb
52:      return result.service_failure!(code: 'webhook_error', message: 'Organization not found') unless organization
66:        return result.service_failure!(code: 'webhook_error', message: 'Invalid signature')
78:        return result.service_failure!(

app/services/payment_providers/stripe_service.rb
79:      result.service_failure!(code: 'webhook_error', message: 'Invalid payload')
81:      result.service_failure!(code: 'webhook_error', message: 'Invalid signature')

app/services/billable_metrics/prorated_aggregations/sum_service.rb
40:        result.service_failure!(code: 'aggregation_failure', message: e.message)
95:        result.service_failure!(code: 'aggregation_failure', message: e.message)

app/services/billable_metrics/aggregations/sum_service.rb
28:        result.service_failure!(code: 'aggregation_failure', message: e.message)
64:        result.service_failure!(code: 'aggregation_failure', message: e.message)

app/services/billable_metrics/aggregations/max_service.rb
19:        result.service_failure!(code: 'aggregation_failure', message: e.message)
42:        result.service_failure!(code: 'aggregation_failure', message: e.message)

app/services/billable_metrics/aggregations/latest_service.rb
19:        result.service_failure!(code: 'aggregation_failure', message: e.message)
43:        result.service_failure!(code: 'aggregation_failure', message: e.message)

app/services/payment_provider_customers/adyen_service.rb
34:      result.service_failure!(code: adyen_error.code, message: adyen_error.msg) unless adyen_success

app/services/payment_provider_customers/stripe_service.rb
39:      result.service_failure!(code: 'stripe_error', message: e.message)
83:      result.service_failure!(code: 'stripe_error', message: e.message)
```

Reply:
Interesting 🤔

Reply:
We do have an `error_code`, and sometimes for service_failure a human readable message. This message is not supposed to be exposed to the end user directly, it is more intended to be returned in the response of the REST API.

The front-end usually have some logic to handle the error code